### ISSUE TITLE
Issues with ReferenceMoistAir (isentropicEnthalpy,  Inverses.p_dTX

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1217]
**Reported by otter on 18 Jul 2013 07:32 UTC**
I have included in ModelicaTest new test models for media that call all media functions at the reference point. For ReferenceMoistAir two issues have been detected:

**Issue 1:**
When simulating ModelicaTest.Media.TestAllProperties.IncompleteMedia.ReferenceMoistAir, Dymola reports the following error:
```
RealMatchingSizes(a,b)
The following error was detected at time: 0
Array dimensions did not match
The stack of functions is:
isentropicEnthalpy_Unique18
isentropicEnthalpy_Unique18(
202650, 
state)
```

**Issue 2:**
If issue 1 is fixed (or the statement Medium.SpecificEnthalpy h_is=Medium... is commented out in model ModelicaTest.Media.TestAllProperties.IncompleteMedia.PartialMediumFunctionsForRealCondensingGases), and model ModelicaTest.Media.TestAllProperties.IncompleteMedia.ReferenceMoistAir, is simulated, Dymola reports the following error:
```
The following error was detected at time: 0
The arguments u_min and u_max provided in the function call
    solveOneNonlinearEquation(f,u_min,u_max)
do not bracket the root of the single non-linear equation 0=f(u):
  u_min  = 173.15
  u_max  = 2000
  fa = f(u_min) = -1.17488
  fb = f(u_max) = -1.15367
fa and fb must have opposite sign which is not the case
The stack of functions is:
Modelica.Math.Nonlinear.solveOneNonlinearEquation
Modelica.Media.Air.ReferenceMoistAir.Utilities.Inverses.p_dTX
Modelica.Media.Air.ReferenceMoistAir.Utilities.Inverses.p_dTX(d, 298.15, {0.01, 0.99})
```
The reason is that function
Modelica.Media.Air.ReferenceMoistAir.Utilities.Inverses.p_dTX has a wrong interval for the inverse (seems to be a copy bug, by using the interval for "T" instead of the interval of "p"):
```
  p := Modelica.Math.Nonlinear.solveOneNonlinearEquation(
          function p_dTX_res(
            d=d,
            T=T,
            X=Xfull),
          173.15,    // should be 0
          2000.0,    // should be 1e6 or something like that
          1e-9);
```

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1217

**Comment by modelica-trac-importer** [comment]
**Comment by wischhusen on 18 Jul 2013 11:00 UTC**
Changed internal calculation of X in isenthalpicEnthalpy function:


```
X := refState.X;
```

and changed the inverse interval in function Modelica.Media.Air.ReferenceMoistAir.Utilities.Inverses.p_dTX to the complete range of the medium model:


```
  p := Modelica.Math.Nonlinear.solveOneNonlinearEquation(
          function p_dTX_res(
            d=d,
            T=T,
            X=Xfull),
          611.2,    
          1e7,    
          1e-9);
```

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 18 Jul 2013 11:36 UTC**
Fixed in 2fdd36de6e3ce0efaf1b5b97b4b215094d4f07a8 and a5894ae353492714fbb2f99d25b6504a9d014d6e

**Comment by modelica-trac-importer** [comment]
**Modified by otter on 20 Dec 2016 15:13 UTC**
